lib/smooth/core/network/Wifi.cpp

```diff
@@ -194,23 +200,54 @@ namespace smooth::core::network
         std::stringstream mac;
 
         uint8_t m[6];
-
-        if (esp_wifi_get_mac(WIFI_IF_STA, m) == ESP_OK)
+        
+        wifi_mode_t mode;
+        if(esp_wifi_get_mode(&mode) == ESP_OK) 
         {
-            for (const auto& v : m)
+            esp_err_t ret;
+            if(mode == WIFI_MODE_STA) 
+                ret = esp_wifi_get_mac(WIFI_IF_STA, m);
+            else
+                ret = esp_wifi_get_mac(WIFI_IF_AP, m);
+
+            if (ret == ESP_OK)
             {
-                if (mac.tellp() > 0)
+                for (const auto& v : m)
                 {
-                    mac << "_";
-                }
+                    if (mac.tellp() > 0)
+                    {
+                        mac << "_";
+                    }
 
-                mac << std::hex << static_cast<int>(v);
+                    mac << std::hex << static_cast<int>(v);
+                }
```

Comment:
Have a look if you can use std::fmt instead of a stringsteam to build the string-version of the mac-address. I believe it is less stack-intense.

Reply:
AFAIK, this is code originally written by you. But I can try to modify it accordingly.

Reply:
Ha, right  you are. I'll have a look at it :)